content/blog/eng/automation-testing-a-step-back.md

```diff
@@ -0,0 +1,114 @@
+
+So we write automated tests to save time in the future, it's an investment. We suppose that when we add more features the old ones need to be checked again, and we invest our time in order to save more in the future, but in the short term we will not have a return.
+
+If we imagine a big project with many developers it becomes more clear, we cannot check manually everything every time (if we want to release often or with simplicity). So if we want to release soon and we have no automation tests we end up testing only what seems most important at the moment, exposing ourselves to the risk of regression.
+
+## Count as documentation
+
+This is an easy point. Documentation can be important (less than working software, see agile manifesto) but it's a pain for most and takes a lot of time to do it well. But if tests are written in a certain way they can have the same information of product documentation, that evolves and adapts as the software grows. This way the software can be self-documented.
+
+## Help write better code
+
+This is a very generic sentence, what do I mean by "better code"? First say that software as every other built thing has a quality. A program can be better than another that does the same things based on some parameters (performance, usability and many others), we can divide these parameters into 2 big groups: the one that users notice and the one that users don't.
+
+For example, if the same code is written without naming convection or indentation the end-users don't notice it and the software runs exactly the same, so why should we care? Because if we want to modify this code, adding new features for example, it will be much much harder.
+
+That said a code is "better" than another when, even if does the same things, it is easier to maintain and to evolve (being more readable, respecting the SOLID principles, etc...).
+
+Tests help to increase quality, because making testable code forces us to restructure it and to change dependency management. This generally translates into a more correct distribution of responsibilities. So it's better from an architectural point of view.
+
+This doesn't mean that untested code have worse distribution of responsibilities. But testing can help to achieve this goal putting us in a position where we need to rethink our code structure.
+
+## Find bug during development
+
+I think it's uncommon but can happen to find a bug out of an automated test during development. It happens because usually the tested cases in automated tests are more structured than manual tests. Precisely because we are forced to write them and we know that they will remain written, it happens that we pay more attention to them. A developer rarely writes all the manual test cases to be performed and then runs them based on how he wrote them. For this reason, the process of writing an automatic test spots problems that may go unnoticed during a manual test.
```

Comment:
I think it is uncommon, but it can happen to find a bug in an automated test during development. It is uncommon because test cases in automated tests are usually more structured than those in manual tests. This is because we are forced to write down automated test cases and we know that they will remain written. Therefore, we pay more attention to automated test cases.